### ISSUE TITLE
Accessibility enhancements

### ISSUE THREAD

**Issue by polx** [xwiki-labs/cryptpad#735]
**Describe the bug**
I've tried to guide a person using only a screenreader to use CryptPad and there was just a few stumble-blocks which prevented a regular usage. These were:

- drive-icons not choosable
- bell and personal-user-icons are not labelled in an understandable manner
- the social button in the participants were not labelled
- overall: there seems to be a lot of labels in some forms of bubbles however, they are not stored in an attribute that is picked by a screen-reader
- some sections of the UI are not labelled

**Where did it happen?**
We tried cryptpad.fr using an edge browser, I was in zoom looking at the screen.

**To Reproduce**
1. Go to cryptpad.fr, reach the drive
2. Using the screen-reader, navigate to the Documents or Recents icons
3. Can't click on them using the accessibility tools: there is not <a> or <button> elements (or <input> or...). The only choice is to use a simulated mouse which is cumbersome and requires big handiness,
4. Once selected... do the same with drive's elements... can't click on them using the accessibility tools. Need to go to use the simulated mouse (same reason).
5. Open a code pad.
6. Sharing worked
7. Changing title: Pen button seemed to have no label that made it recognizable (telling to click worked)
8. Next in row was the bell icon then the lower-case "l" (my user's name's first letter): that "l" is the label of the button... but for a beginner this is quite an unclear function (for sighted people too btw): a label "user-name: l..." would have done a lot. 
9. next in row are the edit panes and preview panes: they are not labelled to indicate where you land, but the user could feel the input zone and a differentiation of preview and code looking at the syntax (e.g. no "#" on the preview).
10. Next in row are then the participants' view which carry no label and, especially, the buttons to either edit the participants' name  or friends: they are not labelled either.

**Expected behavior**
Labels.
But... after having a guide... the conclusion was that the tool seemed usable, at least if the learning remains...

**Screenshots**
<nothing in sight>

**Browser (please complete the following information):**
* Microsoft Windows Version 21H1 (OS Build 19043.985)
* Microsoft Edge Version 90.0.818.66
* JAWS version 2021.2105.53.400 ([JAWS](https://support.freedomscientific.com/Downloads/JAWS) is the screen reader) 
* Extensions installed:  none.

**Additional context**

**Comment by bakeratsby** [comment]
Came across this trying to fill out a form. To specify my experience:

- When navigating forms via keyboard users are unable to select checkboxes or radio buttons via Return key.
- Options and their respective checkbox or radio button are not properly described for screen readers. It is not clear which radio button or checkbox selection belongs to which provided text option when navigating back and forth through the form.
- In a matrix the options per column are not described per row. It is unclear which selection a screen reader user could make (if they were able to via return key).

**Comment by aneverhopefulhelper** [comment]
I have friends who would like to use cryptpad, however, they cannot as it is not accessible at all as it is not tabbable i.e. they cannot tab through the various parts of the UI. It would be great if you could fix this and the issues above, thank you!

**Comment by RastislavKish** [comment]
Hello,

first of all, thank you all for working on this, Cryptpad is an amazing project in the FOSS community, and making it accessible will not only bring it to a larger audience, but also make the project credible for use in public organizations and institutions requiring their materials and software to be universally accessible and could thus bring more recognition.

Is there any specific strategy / roadmap of implementing accessibility in Cryptpad? I'm a blind user and would love to help out reporting various issues, but the project is quite large, consisting of several applications, each with its own unique accessibility situation.

In some, like the Cryptpad Form, there are mostly minor, easy to fix issues such as unlabeled buttons, missing the label tags for binding Form questions to the answer elements, or using non-semantic HTML elements for purposes where semantic elements exist, e.g. using a span with attached click event instead of a button or anchor, not using combo-boxes / radio boxes for things like selecting the question type in the form editor (currently there seems to be a custom component inaccessible for the screenreaders).

In Cryptpad Document, I'm for some reason unable to use the richtext editor, even though CKEditor (if this is the editor used) claims to be WCAG compliant. I'm not seeing this problem in Code, which I guess uses CodeMirror, there are some glitches but I can work with the text just fine.

I'm very happy this is being worked on, but I also understand the project is large, so if there is any roadmap for the a11y work, I would like to follow it to provide the best assistance.

**Comment by polx** [comment]
Hello @RastislavKish ,

really cool to hear that a blind person is successful at use of some parts of Cryptpad.
I am not blind but I helped one to work with it in the Math Working group fo the W3C. Same as you said, we are successful with the code editor (and this since years).

How to report a bug? I think that it is as simple as you do here. At best with separate objectives:

- labelling/alt-tags (these are the easy ones), possibly with changed element names (that's a bit more convoluted)
- separting by app (but considering that the drive and the choice of document type is a separate app)
- particularly open for the rich-text editor which, for some unknown reason does not work (I guess here, doing a meeting with looking over the shoulder is the only method).

I'll leave someone of the team confirm these practices.
I believe that timelines here are not easily fixable at least as validation takes particular persons to be complete.

**Comment by dariiing** [comment]
Hello @RastislavKish and @polx,
Thanks for reaching out. We're currently trying to fix issues by their severity, having a particular focus on keyboard and screen reader usage. We'd be more than happy to receive any feedback or issues that you find, you can contact us through the various channels we have. The easiest way to get into contact is by Email (`a11y@cryptpad.org`) , [Forum](https://forum.cryptpad.org/t/accessibility),  [Matrix chat ](https://matrix.to/#/#cryptpad-accessibility:matrix.xwiki.com).

We keep things organized by tracking issues in our [Accessibility project](https://github.com/orgs/cryptpad/projects/5). If you’d like to help by reporting something, you can either open a new issue yourself (no need to worry about labels, we’ll handle that), or just reach out to us through any of our other channels and we’ll create it for you, whatever works best for you. In the project, you can check the `In Progress` column to see what we’re currently working on for upcoming releases, and the `Done` column to see what has already been implemented/fixed.

Regarding the Forms problems, I'm happy to announce that this is something that I'm [currently working on](https://github.com/cryptpad/cryptpad/pull/1923), and the fix is planned for the Autumn release. As for third-party apps, there are some limitations on our end that we unfortunately can't fix. You can read more about this in [our documenation](https://docs.cryptpad.org/en/user_guide/accessibility.html).

Thanks again for offering to help, it really means a lot! If you'd like to keep the conversation going, feel free to reach out through one of the channels I previously mentioned.

**Comment by dariiing** [comment]
Hello again,
We're closing this issue as it covers too many different topics to be addressed as a whole. Some of the problems mentioned here have already been resolved, and the remaining points are similar to issues we're already tracking. These are being addressed through existing tickets, which you can find in the [Accessibility project](https://github.com/orgs/cryptpad/projects/5).

If you encounter any new issues, please open a separate ticket for each one, or reach out to us through the channels mentioned in the other comment, all listed in our [documentation](https://docs.cryptpad.org/en/user_guide/accessibility.html).

Don't hesitate to reach out, we truly value your input and appreciate your help, we're always open to any feedback you might have. Thank you again! :)
